packages/core/core-flows/src/order/workflows/delete-order-payment-collection.ts

```diff
@@ -0,0 +1,47 @@
+import { PaymentCollectionDTO } from "@medusajs/types"
+import { MedusaError, Modules, PaymentCollectionStatus } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+} from "@medusajs/workflows-sdk"
+import { removeRemoteLinkStep, useRemoteQueryStep } from "../../common"
+
+/**
+ * This step validates that the order doesn't have an active payment collection.
+ */
+export const throwUnlessStatusIsNotPaid = createStep(
+  "validate-payment-collection",
+  ({ paymentCollection }: { paymentCollection: PaymentCollectionDTO }) => {
+    if (paymentCollection.status !== PaymentCollectionStatus.NOT_PAID) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        `Can only delete payment collections where status is not_paid`
+      )
+    }
+  }
+)
+
+export const deleteOrderPaymentCollectionsId =
+  "delete-order-payment-collectionworkflow"
+/**
+ * This workflow deletes one or more invites.
+ */
+export const deleteOrderPaymentCollections = createWorkflow(
+  deleteOrderPaymentCollectionsId,
+  (input: WorkflowData<{ id: string }>): WorkflowData<void> => {
+    const paymentCollection = useRemoteQueryStep({
+      entry_point: "payment_collection",
+      fields: ["id", "status"],
+      variables: { id: input.id },
+      throw_if_key_not_found: true,
+      list: false,
+    }).config({ name: "payment-collection-query" })
+
+    throwUnlessStatusIsNotPaid({ paymentCollection })
+
+    removeRemoteLinkStep({
```

Comment:
question: Shouldn't we delete the actual payment collection as well? 🤔

Reply:
The link has cascade enabled in this PR that should take care of deleting the payment collection

<img width="1056" alt="Screenshot 2024-08-19 at 20 03 39" src="https://github.com/user-attachments/assets/d2f8bfe2-feb1-40e2-aa15-a11295952ad3">

Reply:
Ah, I missed that – that's neat!